File: packages/rust-sdk/src/operations/environments.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use crate::http::substitute_path_params;
use std::sync::Arc;

pub struct EnvironmentsService {
    client: Arc<StraitClient>,
}

impl EnvironmentsService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn list(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/environments", query, None, None).await
    }

    pub async fn create(&self, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("POST", "/v1/environments", None, None, Some(body)).await
    }

    pub async fn get(&self, env_id: &str) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/environments/{envID}", &[("envID", env_id)]);
        self.client.do_request("GET", &path, None, None, None).await
    }

    pub async fn update(&self, env_id: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/environments/{envID}", &[("envID", env_id)]);
        self.client.do_request("PATCH", &path, None, None, Some(body)).await
    }

    pub async fn delete(&self, env_id: &str) -> Result<(), StraitError> {
        let path = substitute_path_params("/v1/environments/{envID}", &[("envID", env_id)]);
        self.client.do_request_no_content("DELETE", &path, None, None, None).await
    }

    pub async fn list_variables(&self, env_id: &str) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/environments/{envID}/variables", &[("envID", env_id)]);
        self.client.do_request("GET", &path, None, None, None).await
    }
}

File: packages/go-sdk/operations/environments.go
package operations

import "context"

// EnvironmentsService provides environment management operations.
type EnvironmentsService struct{ r Requester }

// NewEnvironmentsService creates a new EnvironmentsService.
func NewEnvironmentsService(r Requester) *EnvironmentsService {
	return &EnvironmentsService{r: r}
}

func (s *EnvironmentsService) List(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/environments", query, nil, nil, &result)
	return result, err
}

func (s *EnvironmentsService) Create(ctx context.Context, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", "/v1/environments", nil, nil, body, &result)
	return result, err
}

func (s *EnvironmentsService) Get(ctx context.Context, envID string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", PathParams("/v1/environments/{envID}", map[string]string{"envID": envID}), nil, nil, nil, &result)
	return result, err
}

func (s *EnvironmentsService) Update(ctx context.Context, envID string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "PATCH", PathParams("/v1/environments/{envID}", map[string]string{"envID": envID}), nil, nil, body, &result)
	return result, err
}

func (s *EnvironmentsService) Delete(ctx context.Context, envID string) error {
	return s.r.DoRequest(ctx, "DELETE", PathParams("/v1/environments/{envID}", map[string]string{"envID": envID}), nil, nil, nil, nil)
}

func (s *EnvironmentsService) ListVariables(ctx context.Context, envID string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", PathParams("/v1/environments/{envID}/variables", map[string]string{"envID": envID}), nil, nil, nil, &result)
	return result, err
}
